paper-radio-button.js

```diff
@@ -180,10 +185,57 @@ Polymer({
         /* slightly darker than the button, so that it's readable */
         opacity: 0.65;
       }
+
+      /* When the element is in 'icon' mode rather than classic radio buttons,
+       * change styles
+       */
+      :host([icon]) #radioContainer {
+        width: 24px;
+        height: 24px;
+      }
+
+      :host([icon]) #offRadio, :host([icon]) #onRadio {
+        transition: color ease 0.28s;
+      }
+
+      :host([icon]) #offRadio {
+        border: none;
+        background-color: initial;
+        transition: initial;
+      }
+
+      :host([icon]) #onRadio {
+        background-color: initial;
+        -webkit-transform: initial;
+        transform: initial;
+        transition: initial;
+        will-change: initial;
+      }
+
+      :host([icon][checked]) #offRadio {
+        border-color: initial;
+        color: var(--paper-radio-button-checked-color, var(--primary-color));
+      }
+
+      :host([icon][checked]) #onRadio {
+        -webkit-transform: initial;
+        transform: initial;
+      }
+
+      :host([icon][checked]) #offRadio {
+        border-color: initial;
+      }
+
+      :host([icon][checked]) #onRadio {
+        background-color: initial;
+        color: var(--paper-radio-button-unchecked-color, var(--primary-text-color));
+      }
     </style>
 
     <div id="radioContainer">
-      <div id="offRadio"></div>
+      <div id="offRadio">
+        <slot name="icon"></slot>
```

Comment:
what about putting the slot on `#radioContainer` instead have have `#onRadio` `#offRadio` as the fallback content. That would allow you not override any of the existing CSS rules.

Reply:
Something like

```
<slot name="icon">
  <p>I'm some default content!</p>
</slot>
```
One challenge I ran into making this PR is that it seems hard to actually change the styles in shadow DOM from outside the stylesheet. I couldn't apply changes to `#offRadio` unless it was in this element or I exposed a variable.

Reply:
I as more thinking of:
```
<div id="radioContainer">
   <slot name="radioContent">
       <div id="onRadio"></div>
       <div id="offRadio"></div>
   </slot>
</div>
```
And the CSS for `#onRadio`  and `#offRadio` would get automatically trashed, if you override the slot.

Reply:
If I put something custom in the slot, can I customize it using external CSS? Or is it still bound by shadow DOM rules?

Reply:
Assuming it's an `iron-icon` and the existing CSS rules that applied on `#onRadio` and `#offRadio` are discarded, would it need any additional external CSS than the one included in the `<iron-icon>` shadow DOM?

Reply:
I would want the icon to change color based on whether the paper-radio-button is checked, as it does now.

Reply:
https://developer.mozilla.org/en-US/docs/Web/CSS/::part or have a custom `iron-radio-selectable` that could be used with https://www.webcomponents.org/element/@polymer/paper-radio-group ?

Reply:
For our purposes we really just need to stick an icon in there. Maybe we can use the `::part` rule, but I don't want to implement too much custom behavior.

Reply:
I'd be interested in getting some advice from @bicknellr on what may be the most idiomatic way

Reply:
I've updated the PR to use `::part` and extract the style behavior outside of the component. I'm not sure if this is the best approach, as it means the default will not work very well, but it should be otherwise be functional.

Reply:
Now that you're using `::part` that should mean that the CSS could live outside of the element, no? So the only patch you really need here is a `<slot>` (and putting in on the parent would allow you to discard both element and their associated css rule).